transport-native-epoll/src/main/c/io_netty_channel_epoll_Native.c

```diff
@@ -1403,47 +1403,88 @@ JNIEXPORT jint JNICALL Java_io_netty_channel_epoll_Native_connectDomainSocket(JN
     return 0;
 }
 
-JNIEXPORT jint JNICALL Java_io_netty_channel_epoll_Native_recvFd(JNIEnv* env, jclass clazz, jint fd) {
-    jint socketFd;
-    struct msghdr msg;
-    struct iovec iov[1];
-    struct cmsghdr* ctrl_msg = NULL;
-    char msg_buffer[1];
-    char elem_buffer[CMSG_SPACE(sizeof(int))];
-
-    /* Fill all with 0 */
-    memset(&msg, 0, sizeof(struct msghdr));
-    memset(elem_buffer, 0, CMSG_SPACE(sizeof(int)));
-
-    iov[0].iov_base = msg_buffer;
-    iov[0].iov_len = 1;
-
-    msg.msg_iov = iov;
-    msg.msg_iovlen = 1;
-    msg.msg_control = elem_buffer;
-    msg.msg_controllen = CMSG_SPACE(sizeof(int));
-
-    if(recvmsg(fd, &msg, MSG_CMSG_CLOEXEC) < 0) {
-        // All read, return -1
-        return -1;
-    }
+JNIEXPORT jint JNICALL Java_io_netty_channel_epoll_Native_recvFd0(JNIEnv* env, jclass clazz, jint fd) {
+    int socketFd;
+    struct msghdr descriptorMessage = { 0 };
+    struct iovec iov[1] = { 0 };
+    char control[CMSG_SPACE(sizeof(int))] = { 0 };
+    char iovecData[1];
+
+    descriptorMessage.msg_control = control;
+    descriptorMessage.msg_controllen = sizeof(control);
+    descriptorMessage.msg_iov = iov;
+    descriptorMessage.msg_iovlen = 1;
+    iov[0].iov_base = iovecData;
+    iov[0].iov_len = sizeof(iovecData);
 
-    if((msg.msg_flags & MSG_CTRUNC) == MSG_CTRUNC) {
-      // Not enough space ?!?!
-      return -1;
-    }
+    ssize_t res;
+    int err;
+
+    for (;;) {
+        do {
+            res = recvmsg(fd, &descriptorMessage, 0);
+            // Keep on reading if we was interrupted
+        } while (res == -1 && ((err = errno) == EINTR));
+
+        if (res == 0) {
+            return 0;
+        }
+
+        if (res < 0) {
+            return -err;
```

Comment:
Perhaps we could merge the two `if`s:

    if (res <= 0) {
        return -err;
    }